src/main/java/at/hannibal2/skyhanni/features/misc/TabWidgetSettings.kt

```diff
@@ -0,0 +1,63 @@
+package at.hannibal2.skyhanni.features.misc
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiContainerEvent
+import at.hannibal2.skyhanni.utils.InventoryUtils.getInventoryName
+import at.hannibal2.skyhanni.utils.InventoryUtils.getUpperItems
+import at.hannibal2.skyhanni.utils.ItemUtils.getLore
+import at.hannibal2.skyhanni.utils.LorenzColor
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.ContainerChest
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+class TabWidgetSettings {
+
+    private val mainPageWidgetPattern = "^§7Currently:.*".toRegex()
+
+    private val subPageWidgetPattern = "^§eClick to .*".toRegex(RegexOption.IGNORE_CASE)
+
+    private val shownSettingPattern = "Shown .* Setting.*".toRegex(RegexOption.IGNORE_CASE)
```

Comment:
Use RepoPatterns instead of Regex. I think it is your first time working with RepoPattern so please give the top level documentation a look.
To solve the IngoreCase just make the inventoryName.uppercase()
The normal way to do it:
```Kotlin
    private val mainPageWidgetPattern by RepoPattern.of("tab.widget.setting.main","^§7Currently:.*")
    private val subPageWidgetPattern by RepoPattern.of("tab.widget.setting.sub","^§eClick to .*")
    private val shownSettingPattern by RepoPattern.of("tab.widget.setting.show","Shown .* Setting.*")
```
But since you have multiple patterns you should use a repoGroup
```suggestion
    private val repoGroup = RepoPattern.group("tab.widget.setting")
    private val mainPageWidgetPattern by repoGroup.of("main","^§7Currently:.*")
    private val subPageWidgetPattern by repoGroup.of("sub","^§eClick to .*")
    private val shownSettingPattern by repoGroup.of("show","Shown .* Setting.*")
```
sorry if the syntax is a bit off, since I wrote it in plain text